sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -644,7 +643,21 @@ public CosmosClientBuilder multipleWriteRegionsEnabled(boolean multipleWriteRegi
      * @return current CosmosClientBuilder
      */
     public CosmosClientBuilder clientTelemetryEnabled(boolean clientTelemetryEnabled) {
-        this.clientTelemetryConfig.sendClientTelemetryToServiceEnabled(clientTelemetryEnabled);
+        ImplementationBridgeHelpers.CosmosClientTelemetryConfigHelper.CosmosClientTelemetryConfigAccessor accessor =
```

Comment:
hmm, what if we just track the flag in this step -> 
```
private boolean explicitlyClientTelemetryEnabled;
public CosmosClientBuilder clientTelemetryEnabled(boolean clientTelemetryEnabled) {
      explicitlyClientTelemetryEnabled = clientTelemetryEnabled;
}
```

same for clientTelemetryConfig

and then in clientBuilder.build() --> we kind combine them together

Reply:
Order of calls is important - check the unit test I added - that will explain